### DISCUSSION TITLE
NavigationStack parent doesn't see child changes

### DISCUSSION THREAD

**Discussion by TC72** [pointfreeco/swift-composable-architecture#2929] [Q&A] [answered]
I'm adding navigation to my app but parents are no longer seeing changes in the child.

I started with a basic structure of days > day > note using IdentifiedArrayOf to handle the collections.
This was all working using basic views that just displayed everything and I could add days and notes and use a json file to save my days. I could restart the app, read the json and get back to exactly what I'd saved.

But for the real app I need some kind of navigation to move through the list of days and get to the notes to edit.
I have the navigation working now but changes I make in the child no longer move up to the parent.

I've seen the SyncUps example and that uses delegates to make changes in the parent. Is this the only way for this to work when using NavigationStacks? My app is quite simple and I may just use a simple state to decide which day and note to display if delegates are needed. Seems to spoil the simplicity of child parent connections in TCA.

**Comment by mbrandonw** [comment, marked as answer]
The way sync ups is on `main` right now is one option for sharing state between multiple features. We do have a currently running [beta](https://github.com/pointfreeco/swift-composable-architecture/discussions/2857) that introduces more shared state tools, and the [sync ups](https://github.com/pointfreeco/swift-composable-architecture/tree/shared-state-beta/Examples/SyncUps) app is refactored to use those tools. We are also running a current series of episodes that discusses these tools in depth.

**Reply by TC72** [reply]
Thanks, I had seen what was going on with the state tools. I didn't think it fitted with what I was doing but this shows me it probably does. I'll check out the updated syncups in the state beta branch.
